src/SwedbankPay.Sdk.Infrastructure/JsonSerialization/Converters/CustomMetaDataConverter.cs

```diff
@@ -0,0 +1,74 @@
+﻿using SwedbankPay.Sdk.Common;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace SwedbankPay.Sdk.JsonSerialization.Converters
+{
+    public class CustomMetaDataConverter : JsonConverter<MetadataResponse>
+    {
+        public override MetadataResponse Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException();
+            }
+
+            var metadata = new MetadataResponse();
+
+            while (reader.Read())
+            {
+                if (reader.TokenType == JsonTokenType.EndObject)
+                {
+                    return metadata;
+                }
+
+                string keyString = reader.GetString();
+                reader.Read();
+
+                if(reader.TokenType == JsonTokenType.String)
+                {
+                    string itemValue = reader.GetString();
+
+                    if (keyString.Equals(nameof(MetadataResponse.Id)))
```

Comment:
Codacy found an issue: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-dotnet/pullRequest?prid=6290774)